pkg/oidc/configure.go

```diff
@@ -0,0 +1,77 @@
+// everest
+// Copyright (C) 2023 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package oidc ...
+package oidc
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"github.com/percona/everest/pkg/common"
+	"github.com/percona/everest/pkg/kubernetes"
+)
+
+// OIDC describes the command to configure OIDC settings.
+type OIDC struct {
+	config     Config
+	kubeClient *kubernetes.Kubernetes
+	l          *zap.SugaredLogger
+}
+
+// Config stores configuration for the OIDC command.
+type Config struct {
+	// KubeconfigPath is a path to a kubeconfig
+	KubeconfigPath string `mapstructure:"kubeconfig"`
+	// IssuerURL OIDC issuer url.
+	IssuerURL string `mapstructure:"issuer-url"`
+	// ClientID ID of the client OIDC app.
+	ClientID string `mapstructure:"client-id"`
+}
+
+// NewOIDC returns a new OIDC struct.
+func NewOIDC(c Config, l *zap.SugaredLogger) (*OIDC, error) {
+	kubeClient, err := kubernetes.New(c.KubeconfigPath, l)
+	if err != nil {
+		return nil, err
+	}
+
+	cli := &OIDC{
+		config:     c,
+		kubeClient: kubeClient,
+		l:          l,
+	}
+	return cli, nil
+}
+
+// Run runs the command.
+func (u *OIDC) Run(ctx context.Context) error {
+	err := u.kubeClient.UpdateEverestSettings(ctx, common.EverestSettings{
+		OIDCIssuerURL: u.config.IssuerURL,
+		OIDCClientID:  u.config.ClientID,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = u.kubeClient.RestartDeployment(ctx, common.PerconaEverestDeploymentName, common.SystemNamespace)
```

Comment:
I think we should in some way inform the user that we will be restarting the Everest server. Or maybe even better, allow the user to skip the restart and let them do it themselves later.

For example, once the ConfigMap is configured, maybe we can show a wizard, like: `Everest requires a restart for the changes to take effect. Restart now? (y/N): `

And a CLI flag so that they can skip the prompt and always perform a restart, like `-r` or `-f` (for force). Example:
```bash
$ everestctl oidc configure -f ...
```

Reply:
I agree, the wizard option is more user-friendly then the forced restart, thank you for the suggestion. I think then we should also provide a command to restart Everest if the user chooses restart later. 
Let's also involve @recharte to agree on the command design

Reply:
Before I share my opinion on how to approach the restart option I'd like to ask why do we need to restart the API server?

Reply:
@recharte the JWT middleware which validates the OIDC token is configured once, at the time of initialising the HTTP server. So after you add the OIDC config, the middleware needs to be setup again, for which we need to re-initialise the HTTP server, hence the restart..

For reference, this is the code I'm talking about: https://github.com/percona/everest/pull/324/files#diff-0c59c4dd4c8c53065d0ab7e81ef16c96419f65ba10879efa1c8aecb1e0a73f3dR133-R137

Reply:
Alternatively, to sync the config changes without a restart, the middleware will have to Get the configmap on every request, which will cause a performance overhead. However, we can avoid making a request to k8s each time with the help of an informer. For example, we can setup an informer to watch this ConfigMap and store it in an in-memory structure. Then we read it from memory, instead of making a request to k8s each time.

But for the initial implementation, I felt that taking the restart route would be quicker.

Reply:
Thank you, yes, restart is better for now. 

I'd vote for the following:
1. Wizard - We ask for both fields (clientID and issuersURL) and then confirm the restart like Mayank suggested (`Everest requires a restart for the changes to take effect. Restart now? (y/N):`)
2. Headless - We introduce a flag called `--skip-restart`. I think this is clearer than having a `-f` or `-r`. Also, the default (no flag) means that the restart will happen which is what we want.

WDYT?

Reply:
Sounds good to me!

Reply:
UPD: 
after the discussion with Product & Design it was decided to keep the forced restart since the oidc configuration happens mostly once in the Everest lifetime and the restart itself only affects the UI and API.